File: dropwizard/src/test/java/com/yammer/dropwizard/jersey/params/tests/BooleanParamTest.java
package com.yammer.dropwizard.jersey.params.tests;

import com.yammer.dropwizard.jersey.params.BooleanParam;
import org.junit.Test;

import javax.ws.rs.WebApplicationException;
import javax.ws.rs.core.Response;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;
import static org.junit.Assert.fail;

public class BooleanParamTest {
    @Test
    public void trueReturnsTrue() throws Exception {
        final BooleanParam param = new BooleanParam("true");

        assertThat(param.get(),
                   is(true));
    }

    @Test
    public void uppercaseTrueReturnsTrue() throws Exception {
        final BooleanParam param = new BooleanParam("TRUE");

        assertThat(param.get(),
                   is(true));
    }

    @Test
    public void falseReturnsFalse() throws Exception {
        final BooleanParam param = new BooleanParam("false");
        
        assertThat(param.get(),
                   is(false));
    }

    @Test
    public void uppercaseFalseReturnsFalse() throws Exception {
        final BooleanParam param = new BooleanParam("FALSE");
        
        assertThat(param.get(),
                   is(false));
    }

    @Test
    @SuppressWarnings("ResultOfObjectAllocationIgnored")
    public void nullThrowsAnException() throws Exception {
        try {
            new BooleanParam(null);
            fail("expected a WebApplicationException, but none was thrown");
        } catch (WebApplicationException e) {
            final Response response = e.getResponse();
            
            assertThat(response.getStatus(),
                       is(400));
            
            assertThat((String) response.getEntity(),
                       is("\"null\" must be \"true\" or \"false\"."));
        }
    }

    @Test
    @SuppressWarnings("ResultOfObjectAllocationIgnored")
    public void nonBooleanValuesThrowAnException() throws Exception {
        try {
            new BooleanParam("foo");
            fail("expected a WebApplicationException, but none was thrown");
        } catch (WebApplicationException e) {
            final Response response = e.getResponse();

            assertThat(response.getStatus(),
                       is(400));

            assertThat((String) response.getEntity(),
                       is("\"foo\" must be \"true\" or \"false\"."));
        }
    }
}

File: dropwizard-scala_2.9.1/src/test/scala/com/yammer/dropwizard/scala/params/tests/BooleanParamTest.scala
package com.yammer.dropwizard.scala.params.tests

import org.junit.Test
import javax.ws.rs.WebApplicationException
import com.simple.simplespec.Spec
import com.yammer.dropwizard.scala.params.BooleanParam

class BooleanParamTest extends Spec {

  class `A valid boolean parameter` {
    val param = BooleanParam("true")

    @Test def `has a boolean value` = {
      param.value.must(be(true))
    }
  }

  class `An invalid boolean parameter` {
    @Test def `throws a WebApplicationException with an error message` = {
      evaluating {
        BooleanParam("poop")
      }.must(throwAnExceptionLike {
        case e: WebApplicationException => {
          val response = e.getResponse
          response.getStatus.must(be(400))
          response.getEntity.must(be("Invalid parameter: poop (Must be \"true\" or \"false\".)"))
        }
      })
    }
  }

}
